src/main/scala/histogramactors/HistogramMain.scala

```diff
@@ -24,11 +24,11 @@ object HistogramMain extends App {
 class Master(input: String, output: String, system: ActorSystem) extends Actor {
   private val poolSize = 10
 
-  val loader: ActorRef = system.actorOf(Props[Loader](), "loader")
-  val reducer: ActorRef = system.actorOf(Props[Reducer], "reducer")
-  val parserPool: ActorRef = system.actorOf(Props(classOf[ParserPool], poolSize), "parserPool")
-  val mapper: ActorRef = system.actorOf(Props(classOf[Mapper], parserPool, reducer), "mapper")
-  val saver: ActorRef = system.actorOf(Props[Saver](), "saver")
+  val loader: ActorRef = context.actorOf(Props[Loader](), "loader")
+  val reducer: ActorRef = context.actorOf(Props[Reducer], "reducer")
+  val parserPool: ActorRef = context.actorOf(Props(classOf[ParserPool], poolSize), "parserPool")
+  val mapper: ActorRef = context.actorOf(Props(classOf[Mapper], parserPool, reducer), "mapper")
+  val saver: ActorRef = context.actorOf(Props[Saver](), "saver")
   override def receive: Receive = {
     case Start =>
       loader ! Load(input)
```

Comment:
It is possible to start a few jobs. You need to prevent this either redesign it in a way that it's possible to run a few files simultaneously. In another way that would be a complete mess.